### PR TITLE
[NPU] Add ModelSlim MoE prefix regression tests

### LINKED ISSUE #31570: [Feature] [NPU] Implement CI test for different quantization prefixes (w1/2/3; gate/up/down_proj)
### Checklist

- [ ] If this is not a feature request but a general question, please start a discussion at https://github.com/sgl-project/sglang/discussions. Otherwise, it will be closed.
- [ ] Please use English. Otherwise, it will be closed.

### Motivation

To prevent these types of issue https://github.com/sgl-project/sglang/pull/31456 need to support a test that checks different prefixes (w1/gate_proj;w2/up_proj;w3/down_proj); model with w1/2/3 prefixes - https://www.modelscope.cn/models/Eco-Tech/MiniMax-M2.5-w8a8-QuaRot

### Related resources

_No response_

### PR DESCRIPTION
**Pull request by zihanlin-ai**
## Motivation

ModelSlim MoE checkpoints use either `gate_proj/up_proj/down_proj` or MiniMax-style `w1/w3/w2` projection names. PR #31456 added MiniMax support but lacked per-commit regression coverage.

Closes #31570.

## Modifications

- Add a CPU regression test for both projection-name families.
- Assert that they resolve to the fused W13 and W2 schemes.
- Register the test in `base-a-test-cpu`.
- Mock only NPU operator construction so the test runs without `torch_npu`.

No production code is changed.

## Accuracy Tests

In a CI-like CPU environment without `torch_npu`, all 536 suite files were launched and this PR's test passed:

```
{"file": "test/registered/unit/layers/quantization/test_modelslim_moe_prefixes.py", "passed": true, "elapsed": 17}
```

Nine unrelated files failed locally because of the `libz3` ABI or unavailable network access. Negative mutation checks fail as expected, and the test also passed on Ascend A3 with real MiniMax-M2.5 W8A8 metadata.

## Speed Tests and Profiling

Not applicable; this is a CPU-only unit test.

## Checklist

- [x] Format the code with the repository pre-commit hooks.
- [x] Add a unit test registered in per-commit CI.
- [x] Documentation is not affected.
- [x] Accuracy and speed impact are documented.
- [x] Follow the SGLang code style guidance.

## Note for maintainers

Could an authorized maintainer run `/tag-and-rerun-ci`?

---
### CI States

Latest PR Test (Base): :x: [Run #33243667159](https://github.com/sgl-project/sglang/actions/runs/33243667159)
Latest PR Test (Extra): :x: [Run #33243667126](https://github.com/sgl-project/sglang/actions/runs/33243667126)
Latest PR Test (AMD ROCm 7.2): :x: [Run #33243667155](https://github.com/sgl-project/sglang/actions/runs/33243667155)

### CODE REVIEW

File: test/registered/unit/layers/quantization/test_modelslim_moe_prefixes.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+                gate_name=gate_name, up_name=up_name, down_name=down_name
+            ):
+                quant_description = {
+                    f"{prefix}.0.{gate_name}.weight": "W8A8_DYNAMIC",
+                    f"{prefix}.0.{up_name}.weight": "W8A8_DYNAMIC",
+                    f"{prefix}.0.{down_name}.weight": "W8A8_DYNAMIC",
+                }
+                config = ModelSlimConfig(quant_description)
+
+                w13_scheme, w2_scheme = config.get_moe_scheme(torch.nn.Module(), prefix)
```
[the remark this answers is not in the document]
Comment (zihanlin-ai) [reply test/registered/unit/layers/quantization/test_modelslim_moe_prefixes.py:50]: Fixed by mocking only `NPUW8A8Int8MoEMethod` construction. Prefix resolution and scheme selection remain real, and the test still fails if MiniMax naming or W13/W2 mapping regresses.

**Comment by zihanlin-ai** [comment]
The CPU-suite issue is fixed by mocking only NPU operator construction. Could an authorized maintainer run `/tag-and-rerun-ci`?
